src/apple/system.rs

```diff
@@ -283,6 +286,11 @@ impl SystemExt for System {
         }
     }
 
+    #[cfg(all(target_os = "macos", not(feature = "apple-sandbox"), target_arch = "aarch64"))]
+    fn refresh_components_list(&mut self) {
+        self.components = crate::apple::component::temperatures();
```

Comment:
Instead of recreating the `Vec` everytime, could you instead pass it as `&mut` into `temperatures()` where you would then first `clear()` it and then `push` into it. Like that we avoid a few extra allocations.